both/methods.js

```diff
@@ -65,6 +65,11 @@ setSolveTime = function(resultId, solveIndex, solveTime) {
   RoundSorter.addRoundToSort(result.roundId);
 };
 
+function getTodayDateNoTime() {
+  var today = moment();
+  return new Date(Date.UTC(today.year(), today.month(), today.date()));
```

Comment:
`getTodayDateNoTime` is called by the `createCompetition` and `uploadCompetition` methods, both of which are only defined if `Meteor.isServer` is true. These methods are the only places in our entire codebase where `Competitions.insert` occurs (other than in tests). `startDate` is a required attribute of Competition, so it must be specified when creating a competition.

As we discussed before, it's a shame that javascript and mongo don't distinguish between `Date` and `DateWithTime`. I'm not opposed to creating our own "timeless Date" object, but in the meantime, I decided to fix what bugs we have related to this, even though they'll probably keep popping up (timezones are another complication that I'd love to avoid).